ethcontract/src/errors/nethermind.rs

```diff
@@ -0,0 +1,110 @@
+//! This module implements Nethermind specific error decoding in order to try and
+//! provide more accurate errors from Nethermind nodes.
+
+use crate::errors::{revert, ExecutionError};
+use jsonrpc_core::Error as JsonrpcError;
+
+/// Revert error discriminant.
+const REVERTED: &str = "Reverted 0x";
+/// Invalid op-code error discriminant.
+const INVALID: &str = "Bad instruction";
+
+/// Tries to get a more accurate error from a generic Nethermind JSON RPC error.
+/// Returns `None` when a more accurate error cannot be determined.
+pub fn get_encoded_error(err: &JsonrpcError) -> Option<ExecutionError> {
+    let message = get_error_message(err)?;
+    if let Some(hex) = message.strip_prefix(REVERTED) {
+        if hex.is_empty() {
+            return Some(ExecutionError::Revert(None));
+        } else {
+            match hex::decode(hex)
+                .ok()
+                .and_then(|bytes| revert::decode_reason(&bytes))
+            {
+                Some(reason) => return Some(ExecutionError::Revert(Some(reason))),
+                None => return Some(ExecutionError::Revert(None)),
+            }
+        }
+    } else if message.starts_with(INVALID) {
+        return Some(ExecutionError::InvalidOpcode);
+    }
+
+    None
```

Comment:
We can also check:

```
if err.message == "VM execution error" {
    return Some(ExecutionError::Revert(None));
}
```

As a catch-all (in case we missed some other error messages - like what happens when a contract reverts without data?).